### PR TITLE
fix: 400 status response code for csaf.io domain due to missing useragent

### LINKED ISSUE #169: Informative Test 6.3.6
Test 6.3.6 is currently failing. 
Trying out older releases doesn't change this, even though the test worked at the time of releasing.
>6.3.6 valid
   informative/oasis_csaf_tc-csaf_2_0-2021-6-3-06-11.json

I think it is a problem with the "https://csaf.io" domain referenced in the test files. Changing this url makes all tests pass.

But I can't figure out why csaf.io domain is failing the test now, please have a look for yourself @tschmidtb51, maybe you can figure out what is going on here.

### PR DESCRIPTION
**Pull request by MarcoKorinth**
# Changes
- add user agent "csaf-validator-lib/VERSION" to undici requests in testURL module

Fixes #169

### CODE REVIEW

File: lib/informativeTests/shared/testURL.js [hunk 1]
```diff
@@ -1,13 +1,27 @@
...
  * @param {() => void} onError
  */
 export default async function testURL(url, onError) {
+  // set user-agent to csaf-validator-lib/VERSION
+  const userAgent = `${packageInfo.name.split('/').at(-1)}/${packageInfo.version}`
   try {
     const res = await request(url, {
       method: 'HEAD',
+      headers: {
+        'user-agent': userAgent
```
Comment (tschmidtb51) [inline lib/informativeTests/shared/testURL.js:23]: According to the [documentation](https://www.rfc-editor.org/rfc/rfc2616#section-14.43) I checked, `User-Agent` is the correct spelling.
```suggestion
        'User-Agent': userAgent

File: lib/informativeTests/shared/testURL.js [hunk 2]
```diff
@@ -1,13 +1,27 @@
...
 import { request } from 'undici'
 
+/**
+ * @type {{
+ *  name: string
+ *  version: string
+ * }}
+ */
+const packageInfo = createRequire(import.meta.url)('../../../package.json')
+
```
Comment (tschmidtb51) [inline lib/informativeTests/shared/testURL.js:11]: Hmm. I wonder whether we do requests in other places in the library? If not, I'm ok with putting it in this file.

**Review by tschmidtb51 — COMMENTED** [review]
Please see my comments

File: lib/informativeTests/shared/testURL.js (hunk 2 above)
Comment (MarcoKorinth) [reply lib/informativeTests/shared/testURL.js:11]: I searched all files with a fuzzy finder and it seems like the file hasn't been referenced yet.

**Review by tschmidtb51 — APPROVED** [review]
LGTM

**Pull request merged**